harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -102,14 +102,6 @@ def from_trial(*args: Any, **kwargs: Any) -> det.TrialController:
     def from_native(*args: Any, **kwargs: Any) -> det.TrialController:
         raise NotImplementedError("PyTorchTrial only supports the Trial API")
 
-    @staticmethod
-    def support_determined_native() -> bool:
```

Comment:
why are we removing this as part of removing tests?

Reply:
I'm removing it because it's True in all cases.  Also, in the case of `support_determined_native()`, we *actually don't* support it, so it's pretty deceptive.  The code will run but we know it is buggy and not meaningfully useful (hence removing tests).

The related `supports_multi_gpu_training` is simply out-of-date, we've supported that in all cases since we removed TensorFlowTrial and KerasSimpleTrial.

Reply:
I don't think that's what "support" means at all.  If you file a bug that it doesn't work, are we going to respond with "let us fix that", or "sorry we don't actually support that feature, you should stop trying to use it"?

And again, it's True in all cases, so even if it wasn't deceptive, it's not useful.

Reply:
So turns out `supports_determined_native()` is actually talking about the Native API, so I'm leaving it out of this PR entirely.